netbox/netbox_inventory.py

```diff
@@ -71,57 +80,98 @@ def cli():
 
 
 class NetBoxSession(requests.Session):
-    def __init__(self, url, token):
+    """A session for interacting with the NetBox API.
+
+    Attributes:
+        url: The base URL of the NetBox instance.
+        token: The API token for authentication.
+    """
+
+    def __init__(self, url: str, token: str):
+        """Initialize the NetBoxSession.
+
+        Args:
+            url: The base URL of the NetBox instance.
+            token: The API token for authentication.
+        """
         super(NetBoxSession, self).__init__()
         self.url = url
-        self.headers["authorization"] = "Token %s" % token
+        self.headers["authorization"] = f"Token {token}"
         self.verify = False
 
-    def prepare_request(self, request):
+    def prepare_request(self, request: requests.Request) -> requests.PreparedRequest:
+        """Prepare the request by appending the base URL to the request URL.
+
+        Args:
+            request: The request object to prepare.
+
+        Returns:
+            requests.PreparedRequest: The prepared request object.
+        """
         request.url = self.url + request.url
         return super(NetBoxSession, self).prepare_request(request)
 
 
-netbox: NetBoxSession = None
+NETBOX_SESSION: Optional[NetBoxSession] = None
+
 
+@lru_cache
+def get_site(site_slug: str) -> dict:
+    """Retrieve details of a site from NetBox using its slug.
 
-@lru_cache()
-def get_site(site_slug):
-    res = netbox.get("/api/dcim/sites/", params={"slug": site_slug})
+    Args:
+        site_slug: The slug of the site.
+
+    Returns:
+        dict: The site details as a dictionary.
+    """
+    res = NETBOX_SESSION.get("/api/dcim/sites/", params={"slug": site_slug})
     res.raise_for_status()
     return res.json()["results"][0]
 
 
-def create_csv_file(inventory_records, cli_opts):
+def create_csv_file(inventory_records: Iterator[dict], cli_opts: argparse.Namespace) -> None:
+    """Create a CSV file from inventory records.
+
+    Args:
+        inventory_records: An iterator of device records.
+        cli_opts: Parsed command-line options, including the output file.
+    """
     csv_wr = csv.writer(cli_opts.output)
     csv_wr.writerow(CSV_FIELD_NAMES)
 
     for rec in inventory_records:
         csv_wr.writerow(rec_to_csv(rec))
```

Comment:
**Suggestion:** Use a context manager to ensure that the CSV file is properly closed after writing, which is a best practice for file handling. [best practice, importance: 8]
```suggestion
    with open(cli_opts.output, 'w', newline='') as csvfile:
        csv_wr = csv.writer(csvfile)
        csv_wr.writerow(CSV_FIELD_NAMES)
        for rec in inventory_records:
            csv_wr.writerow(rec_to_csv(rec))
```